idl/github.com/uber/cadence/admin.thrift

```diff
@@ -49,3 +59,23 @@ struct DescribeWorkflowExecutionResponse{
   40: optional string mutableStateInCache
   50: optional string mutableStateInDatabase
 }
+
+//At least one of the parameters needs to be provided
+struct DescribeHistoryHostRequest {
+  10: optional string hostAddress //ip:port
+  20: optional i32 shardIdForHost
+  30: optional shared.WorkflowExecution executionForHost
+}
+
+struct DescribeHistoryHostResponse{
+  10: optional i32 numberOfShards
+  20: optional list<i32> shardIDs
+  30: optional DomainCache domainCache
+  40: optional string shardControllerStatus
+  50: optional string address
+}
+
+struct DomainCache{
```

Comment:
It would be more precise to be named DomainCacheInfo or something else

Reply:
That sounds good to me.